internal/storage/db/internal/db.go

```diff
@@ -478,7 +480,97 @@ func (s *dbStorage) GetDependents(ctx context.Context, ids ...namer.ModuleID) (m
 	return out, nil
 }
 
+func (s *dbStorage) GetAncestorsAndDescendants(ctx context.Context, ids ...namer.ModuleID) (map[namer.ModuleID][]namer.ModuleID, error) {
```

Comment:
I think we can make this a little bit more efficient. Right now, for every policy you need to disable, you have to execute two queries and map their values to Go structs. That's 2N queries. Instead, we can run just one query to figure out everything we need to know about the policy set to be disabled.

```sql
SELECT p.id, 
	EXISTS(select 1 FROM policy_ancestor pa1 JOIN policy p1 ON (pa1.policy_id = p1.id AND p1.disabled = false) WHERE pa1.ancestor_id = p.id) AS has_descendants 
FROM policy p 
WHERE p.id IN (?);
```

This will output something like the following:

```
id|has_descendants
-4874326380253481038|0
-311105230672970778|1
1057909666449662682|0
```

If the `has_descendants` column is `0`, we can disable that policy.